src/riscv-double-trap.adoc

```diff
@@ -137,13 +137,13 @@ An `SRET` instruction sets the `SDT` bit to 0.
 
 [NOTE]
 ====
-A trap handler after saving the state needed for resuming from the trap,
-including `scause`, `sepc`, and `stval` among others, should clear the `SDT` bit
-when it is reentrant.
+After a trap handler has saved the state, such as `scause`, `sepc`, 
+and `stval`, needed for resuming from the trap and is reentrant, it 
+should clear the `SDT` bit.
 
-Resetting of the `SDT` by an `SRET` enables the trap handler to detect double
-trap occuring during the tail phase, where it restores critical state to return
-from a trap.
+Resetting the SDT by an SRET enables the trap handler to detect a double 
+trap that occurs during the tail phase, where it restores a critical state 
```

Comment:
@kersten1 "a critical state" -> "critical state" since there are many such critical states that are restored.